protocol/communities/manager.go

```diff
@@ -1681,7 +1681,33 @@ func (m *Manager) CheckPermissionToJoin(id []byte, addresses []gethcommon.Addres
 		}
 		return response, nil
 	}
-	return m.checkPermissions(permissionsToJoin, accountsAndChainIDs, false)
+	checkPermissionsResponse, err := m.checkPermissions(permissionsToJoin, accountsAndChainIDs, false)
+	if err != nil {
+		return nil, err
+	}
+	if checkPermissionsResponse.Satisfied {
+		// If the permission is already satisfied, we can already return
+		return checkPermissionsResponse, nil
+	}
+	if len(becomeMemberPermissions) == 0 {
+		// There is no permission to join, we can just say it's satisfied
+		checkPermissionsResponse.Satisfied = true
+		return checkPermissionsResponse, nil
+	}
+	// If it wasn't satisfied, check if we just satisfy the permission to join the members
+	for permissionId, checkedPermission := range checkPermissionsResponse.Permissions {
+		for _, permission := range becomeMemberPermissions {
+			if permission.Id == permissionId {
+				for _, criteria := range checkedPermission.Criteria {
+					if criteria == true {
+						checkPermissionsResponse.Satisfied = true
+						return checkPermissionsResponse, nil
+					}
+				}
+			}
+		}
+	}
+	return checkPermissionsResponse, nil
```

Comment:
> **If a community has only a permission to become admin**, or a join permission and a permission to become admin, even if the user meets the permission to join, the request will not be sent.

The bold part could be simply fixed by the following:

```diff
diff --git a/protocol/communities/manager.go b/protocol/communities/manager.go
index ecd9f8120..cce86bc56 100644
--- a/protocol/communities/manager.go
+++ b/protocol/communities/manager.go
@@ -1726,7 +1726,7 @@ func (m *Manager) CheckPermissionToJoin(id []byte, addresses []gethcommon.Addres
 	}
 	accountsAndChainIDs := combineAddressesAndChainIDs(addresses, allChainIDs)
 
-	if len(permissionsToJoin) == 0 {
+	if len(becomeMemberPermissions) == 0 || len(permissionsToJoin) == 0 {
 		// There are no permissions to join on this community at the moment,
 		// so we reveal all accounts + all chain IDs
 		response := &CheckPermissionsResponse{
```

--- 

> If a community has only a permission to become admin, **or a join permission and a permission to become admin**, even if the user meets the permission to join, the request will not be sent.

This indicates an issue inside the `checkPermissions`.  `checkPermissions` is permission type agnostic so it shouldn't matter what type of permissions are passed there. It should return `Satisfied == true` if at least one of the permissions is satisfied.

Reply:
> > If a community has only a permission to become admin, or a join permission and a permission to become admin, even if the user meets the permission to join, the request will not be sent.

> This indicates an issue inside the checkPermissions. checkPermissions is permission type agnostic so it shouldn't matter what type of permissions are passed there. It should return Satisfied == true if at least one of the permissions is satisfied.

I tried this scenario again (commented most of my new code) and ran my new tests and they still pass. I just kept the small change you suggested. Maybe there was something else happening when i manually tested or maybe my tests are not good?

Reply:
as an option, try to comment on your changes and run tests. Accordingly to the MR title, `TestJoinCommunityWithAdminPermission` must fail